src/Nethermind/Nethermind.Network/P2P/PacketSender.cs

```diff
@@ -33,6 +33,7 @@ public class PacketSender : ChannelHandlerAdapter, IPacketSender
             }
 
             IByteBuffer buffer = _messageSerializationService.ZeroSerialize(message);
+            message.Dispose();
```

Comment:
should be in some kind of finally or using?

Reply:
Maybe? Hard to know exactly at what point, and at this point, it does not cover all failure point anyway, as in before the message is passed in.

Reply:
Still I think serializer is the riskiest part of this codebase